src/smartcontracts/contract_call.cpp

```diff
@@ -18,5 +18,6 @@ void ContractCall::setArgs(SCArguments args)
 
 std::string ContractCall::asOnData() const
 {
-    return util::stringToHex(m_function) + m_args.asOnData();
+    return (m_function.empty()) ?
```

Comment:
A bit hack-ish (not obvious that `m_args.asOnData()` has a `@` prefix, as well), but can stay as it is for the moment.